src/basic/socket-util.c

```diff
@@ -436,10 +436,14 @@ const char* socket_address_get_path(const SocketAddress *a) {
         return a->sockaddr.un.sun_path;
 }
 
+bool socket_ipv6_is_compiled_in(void) {
+        return access("/sys/module/ipv6", F_OK) == 0;
+}
+
 bool socket_ipv6_is_supported(void) {
         _cleanup_free_ char *l = NULL;
 
-        if (access("/sys/module/ipv6", F_OK) != 0)
+        if (!socket_ipv6_is_compiled_in())
```

Comment:
I'd like `socket_ipv6_is_available()` better, as it is more in line with similar functions in systemd.